### PR TITLE
openbangla-keyboard: 2.0.0 -> unstable-2023-07-21; add fcitx5 support

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

* 2.0.0 -> unstable-2023-07-21
  * needed for fcitx5 support
* fix `meta`
  * remove use of `with lib;` (see #208242)
  * add self to maintainers
* fix IME support (ibus, fcitx5)
* remove existing package in favour of the IME-specific ones
  * Removed `openbangla-keyboard` from top-level, added throws alias for that
  * Added two new packages (`ibus-engines.openbangla-keyboard` and `fcitx5-openbangla-keyboard`)

I see there's a discrepancy with how ibus and fcitx5 handle their packages, but I didn't want to create a pattern that's never been seen before in nixpkgs.

Closes #182388

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hqurve** [comment]
Perhaps the `openbangla-keyboard` derivation should be deprecated (placed in aliases.nix with a message suggesting `ibus-engines.openbangla-keyboard` and `fcitx5-openbangla-keyboard`) since the build fails with

```
 > CMake Error at CMakeLists.txt:79 (message):
>   Please set atleast ENABLE_IBUS or ENABLE_FCITX to be ON to build a backend
```

Besides that, both of `ibus-engines.openbangla-keyboard` and `fcitx5-openbangla-keyboard` build.

**Comment by eclairevoyant** [comment]
Thanks, I completely blanked on the package deprecation process. I'll take a look at that

**Comment by eclairevoyant** [comment]
I've removed the existing attribute, thanks for the suggestion!

**Review by hqurve — APPROVED** [review]
I tested the ibus engine and it works

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -7041,6 +7041,8 @@ with pkgs;
       protobuf = pkgs.protobuf3_21.overrideDerivation (_: { stdenv = clangStdenv; });
     };
 
+    openbangla-keyboard = libsForQt5.callPackage ../applications/misc/openbangla-keyboard { withIbusSupport = true; };
+
```
Comment (thiagokokada) [inline pkgs/top-level/all-packages.nix:7045]: This should be removed.

Comment (eclairevoyant) [reply pkgs/top-level/all-packages.nix:7045]: why?

Comment (thiagokokada) [reply pkgs/top-level/all-packages.nix:7045]: Because you both added a alias throwing this and add this to all-packages.

Unless I understood this wrong.

Comment (thiagokokada) [reply pkgs/top-level/all-packages.nix:7045]: https://github.com/NixOS/nixpkgs/pull/258314/files#diff-29ed0f5fda84c91253503c1664d60a194a324e29518d472adf846f30b8db52e3R598

Comment (eclairevoyant) [reply pkgs/top-level/all-packages.nix:7045]: This is in the `ibus-engines` scope, i.e.
I removed `openbangla-keyboard` from top-level, added throws alias for that, and added two new packages (`ibus-engines.openbangla-keyboard` and `fcitx5-openbangla-keyboard`).

I see there's a discrepancy with how ibus and fcitx5 handle their packages, but I didn't want to create a pattern that's never been seen before in nixpkgs.

Comment (hqurve) [reply pkgs/top-level/all-packages.nix:7045]: This attribute is declared in the `ibus-engines` attribute set. It is not a top level attribute (see line 7024. https://github.com/NixOS/nixpkgs/pull/258314/files#diff-ab5748dc9567516fefba8344056b51ec1866adeace380f46e58a7af3d619ea22R7024-R7045).

Comment (thiagokokada) [reply pkgs/top-level/all-packages.nix:7045]: Oh ok I see, sorry, this is in the `ibus-engines` sub attribute (damn GitHub with its small diffs size).

You are correct here.

**Review by thiagokokada — APPROVED** [review]

**Comment by asifakonjee** [comment]
It does not work with fcitx5.

**Comment by asifakonjee** [comment]
@eclairevoyant... fcitx5 will not work with OpenBangla Keyboard 2.0. It will work with OpenBangla Keyboard built from the develop branch of OpenBangla Keyboard's github repository till now.

**Pull request merged**